src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2397,6 +2397,76 @@ def load_key_and_certificates_from_pkcs12(self, data, password):
 
         return (key, cert, additional_certificates)
 
+    def serialize_key_and_certificates_to_pkcs12(self, name, key, cert, cas,
+                                                 key_encryption):
+        password = None
+        if name is not None:
+            utils._check_bytes("name", name)
+        if key is not None and not isinstance(
+            key, (rsa.RSAPrivateKeyWithSerialization,
+                  dsa.DSAPrivateKeyWithSerialization,
+                  ec.EllipticCurvePrivateKeyWithSerialization)):
+            raise TypeError(
+                "Key must be RSA, DSA, or EllipticCurve private key."
+            )
+        if cert is not None and not isinstance(cert, x509.Certificate):
+            raise TypeError("cert must be a certificate")
+
+        if key_encryption is not None and not isinstance(
+                key_encryption, serialization.KeySerializationEncryption):
+            raise TypeError(
+                "Key encryption algorithm must be a "
+                "KeySerializationEncryption instance"
+            )
+        if isinstance(key_encryption, serialization.NoEncryption):
+            nid_cert = -1
+            nid_key = -1
+            pkcs12_iter = 0
+            mac_iter = 0
+        elif isinstance(key_encryption,
+                        serialization.BestAvailableEncryption):
+            # These are curated values we will (hopefully) update over time
+            nid_cert = self._lib.NID_aes_256_cbc
+            nid_key = self._lib.NID_aes_256_cbc
+            pkcs12_iter = 20000
+            # mac_iter chosen for compatibility reasons, see:
+            # https://www.openssl.org/docs/man1.1.1/man3/PKCS12_create.html
+            mac_iter = 1
+            password = key_encryption.password
+        else:
+            raise ValueError("Unsupported key encryption type")
+
+        if cas is None:
+            sk_x509 = self._ffi.NULL
+        else:
+            cas = list(cas)
+            if not all(isinstance(ca, x509.Certificate) for ca in cas):
+                raise TypeError("cert in cas must be a certificate")
+
+            sk_x509 = self._lib.sk_X509_new_null()
+            sk_x509 = self._ffi.gc(sk_x509, self._lib.sk_X509_free)
```

Comment:
Does this need the `sk_x509_pop_free` thingy?

Reply:
Not sure about it.

Reply:
It does not -- verified via a new memleak test I just pushed. We missed the PKCS12_free though 😄